tests/e2e/nvidia-ctk_containerd_test.go

```diff
@@ -0,0 +1,433 @@
+
+		// Install the NVIDIA Container Toolkit packages
+		_, _, err = installer.Install(nestedContainerRunner)
+		Expect(err).ToNot(HaveOccurred(), "Failed to install toolkit for containerd")
+	})
+
+	AfterAll(func(ctx context.Context) {
+		// Restore original containerd configuration
+		_, _, err := nestedContainerRunner.Run(`
+			# Restore the original config.toml
+			if [ -f /tmp/containerd-config.toml.backup ]; then
+				cp /tmp/containerd-config.toml.backup /etc/containerd/config.toml
+			fi
+			
+			# Restore config.d directory
+			rm -rf /etc/containerd/config.d
+			if [ -d /tmp/containerd-config.d.backup ]; then
+				cp -r /tmp/containerd-config.d.backup /etc/containerd/config.d
+			fi
+			
+			# Restore custom.d directory
+			rm -rf /etc/containerd/custom.d
+			if [ -d /tmp/containerd-custom.d.backup ]; then
+				cp -r /tmp/containerd-custom.d.backup /etc/containerd/custom.d
+			fi
+			
+			# Restart containerd to apply restored config
+			systemctl restart containerd || true
+		`)
+		if err != nil {
+			GinkgoWriter.Printf("Warning: failed to restore containerd configuration: %v\n", err)
+		}
+
+		// Cleanup: remove the container
+		_, _, err = runner.Run(fmt.Sprintf("docker rm -f %s 2>/dev/null || true", containerName))
+		if err != nil {
+			GinkgoWriter.Printf("Warning: failed to cleanup container %s: %v\n", containerName, err)
+		}
+	})
+
+	BeforeEach(func(ctx context.Context) {
+		// Restore from backup to ensure clean state for each test
+		_, _, err := nestedContainerRunner.Run(`
+			# Restore the original config.toml
+			if [ -f /tmp/containerd-config.toml.backup ]; then
+				cp /tmp/containerd-config.toml.backup /etc/containerd/config.toml
+			fi
+			
+			# Clean up and restore config.d directory
+			rm -rf /etc/containerd/config.d
+			if [ -d /tmp/containerd-config.d.backup ]; then
+				cp -r /tmp/containerd-config.d.backup /etc/containerd/config.d
+			else
+				mkdir -p /etc/containerd/config.d
+			fi
+			
+			# Clean up and restore custom.d directory
+			rm -rf /etc/containerd/custom.d
+			if [ -d /tmp/containerd-custom.d.backup ]; then
+				cp -r /tmp/containerd-custom.d.backup /etc/containerd/custom.d
+			fi
+ 
+			# Restart containerd and wait for it to be ready
+			systemctl restart containerd
+			
+			# Wait for containerd to be responsive (up to 30 seconds)
+			for i in $(seq 1 30); do
+				if ctr version >/dev/null 2>&1; then
+					echo "containerd is ready"
+					break
+				fi
+				if [ "$i" -eq 30 ]; then
+					echo "ERROR: containerd failed to become ready"
+					exit 1
+				fi
+				sleep 1
+			done
+		`)
+		Expect(err).ToNot(HaveOccurred(), "Failed to reset containerd configuration")
+	})
+
+	When("configuring containerd on a Kubernetes node", func() {
+		It("should add NVIDIA runtime using drop-in config without modifying the main config", func(ctx context.Context) {
+			// Configure containerd using nvidia-ctk
+			cmd := fmt.Sprintf(nvidiaCtkConfigureContainerdCmd, "--nvidia-set-as-default --cdi.enabled")
```

Comment:
Let's definitely just use string concatenation here. I would even go so far as to say that we should just duplicate the command every time we want to run it.